internal/kafka/internal/handlers/data_plane_cluster_test.go

```diff
@@ -0,0 +1,285 @@
+package handlers
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/private"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/gorilla/mux"
+	. "github.com/onsi/gomega"
+
+	dataplanemocks "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test/mocks/data_plane"
+)
+
+var (
+	fullStatusRequest = `{"Conditions":[{"Type":"Ready","Status":"True"}],"Total":{"IngressEgressThroughputPerSec":"12345Mi","Connections":500,"DataRetentionSize":"12345Mi","Partitions":50},"NodeInfo":{"Ceiling":12,"Floor":3,"Current":6,"CurrentWorkLoadMinimum":6},"Remaining":{"Connections":400,"Partitions":40,"IngressEgressThroughputPerSec":"1234Mi","DataRetentionSize":"1234Mi"},"ResizeInfo":{"NodeDelta":6,"Delta":{"Connections":100,"Partitions":10,"IngressEgressThroughputPerSec":"1234Mi","DataRetentionSize":"1234Mi"}}}`
+)
+
+func Test_UpdateDataPlaneClusterStatus(t *testing.T) {
+	type fields struct {
+		dataplaneClusterService services.DataPlaneClusterService
+	}
+
+	type args struct {
+		body []byte
+	}
+
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		wantStatusCode int
+	}{
+		{
+			name: "should return success status if passing correct and populated DataPlaneClusterStatus",
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{
+					UpdateDataPlaneClusterStatusFunc: func(ctx context.Context, clusterID string, status *dbapi.DataPlaneClusterStatus) *errors.ServiceError {
+						return nil
+					},
+				},
+			},
+			args: args{
+				body: []byte(fullStatusRequest),
+			},
+			wantStatusCode: http.StatusNoContent,
+		},
+		{
+			name: "should return an error if updating dataplane cluster status fails in the service",
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{
+					UpdateDataPlaneClusterStatusFunc: func(ctx context.Context, clusterID string, status *dbapi.DataPlaneClusterStatus) *errors.ServiceError {
+						return errors.GeneralError("failed to update dataplane status")
+					},
+				},
+			},
+			args: args{
+				body: []byte(fullStatusRequest),
+			},
+			wantStatusCode: http.StatusInternalServerError,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewDataPlaneClusterHandler(tt.fields.dataplaneClusterService)
+			req, rw := GetHandlerParams("GET", "/{id}", bytes.NewBuffer(tt.args.body))
+			req = mux.SetURLVars(req, map[string]string{"id": "test-id"})
+			h.UpdateDataPlaneClusterStatus(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
```

Comment:
If expect fails, the Body is not closed. This should be moved before the Expect or `defer` should be used.

Reply:
done